File: MongoButcher/App/Core/Workloads/Resources/Resource.cs
using System.Collections.Generic;
using LeoMongo.Database;
using MongoDBDemoApp.Core.Workloads.ActionHistories;

namespace MongoDBDemoApp.Core.Workloads.Resources
{
    public class Resource : EntityBase
    {
        public string ProductName { get; set; }
        public double Amount { get; set; }
        public List<ActionHistory> ActionHistories { get; set; }
    }
}

File: MongoButcher/Client/src/app/data/resource.ts
import {Product} from './product';
import {ActionHistory} from './actionHistory';

export class Resource {
  id ?: string;
  product ?: Product;
  amount ?: number;
  actionHistories ?: Array<ActionHistory>;

  constructor(id?: string, product?: Product, amount ?: number) {
    this.id = id;
    this.product = product;
    this.amount = amount;
    this.actionHistories = [];
  }
}
